docs/src/views/CustomComponents.tsx

```diff
@@ -0,0 +1,234 @@
+      you want to define some styles or variables for this custom component. Let's see one example:
+    </p>
+
+    <ExampleSnippet
+      value={[
+        `<Provider`,
+        `  theme={{`,
+        `    componentStyles: {`,
+        `      StyledButton: {`,
+        `        root: ({ props, variables, theme: {siteVariables} }) => ({`,
+        `          backgroundColor: siteVariables.brand,`,
+        `          color: (variables as any).color,`,
+        `        }),`,
+        `      },`,
+        `    },`,
+        `    componentVariables: {`,
+        `      StyledButton: {`,
+        `        color: '#F2F2F2',`,
+        `      },`,
+        `    },`,
+        `  }}`,
+        `>`,
+        `  <StyledButton> Provider styled button </StyledButton>`,
+        '</Provider>',
+      ].join('\n')}
+      render={() => (
+        <Provider
+          theme={{
+            componentStyles: {
+              StyledButton: {
+                root: ({ props, variables, theme: { siteVariables } }) => ({
+                  backgroundColor: siteVariables.brand,
+                  color: (variables as any).color,
+                }),
+              },
+            },
+            componentVariables: {
+              StyledButton: {
+                color: '#F2F2F2',
+              },
+            },
+          }}
+        >
+          <StyledButton>Provider styled button</StyledButton>
+        </Provider>
+      )}
+    />
+
+    <p>
+      The second argument of the <code>createComponent</code> config param is the render method.
+      This is the place where you link the Stardust bits with your custom component. This method is
+      invoked with the following parameters:
+    </p>
+    <ul>
+      <li>
+        <code>stardust</code> - this is the object containing the evaluated theming props (classes
+        and rtl)
+      </li>
+      <li>
+        <code>...props</code> - all other props provided by the user
+      </li>
+    </ul>
+    <p>
+      The previous implementation of the render method inside the <code>createComponent</code>{' '}
+      function for the
+      <code>StyledButton</code>, allows the user, to use the styles and variables properties on the
+      custom component in the same way they would be used for any Stardust component.
+    </p>
+
+    <Header as="h2" content="Using the custom components" />
+
+    <p>
+      We already saw how the <code>Provider</code> can define some stylings and variables for the
+      custom components. Next, we will take a look into few examples of how the user can further
+      customize the styling and variables of the custom components, the same way they would do with
+      the Stardust components.
+    </p>
+
+    <Header as="h3" content="Example 1. Using the variables property" />
+
+    <ExampleSnippet
+      value={[
+        `<StyledButton variables={(siteVars) => ({color: siteVars.black})}>`,
```

Comment:
lets try to simplify it a bit - while I do understand that this example is aimed to cover all things that are possible to do with this prop being introduced, at the same time, while being in the client's shoes, I would rather expect something pretty simple that would just deliver necessary idea to me. Otherwise, just imagine that I am not yet familiar with `siteVariables` and what the beast it is - should I learn that or, maybe, it is fine for me to use `variables` without this knowledge? I would say that the later is just fine